File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/OnboardingUserUpdateDto.java
package de.sovity.authorityportal.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Information about the user provided while onboarding.")
public class OnboardingUserUpdateDto {

    @NotBlank(message = "First name cannot be blank.")
    @Schema(description = "First name", requiredMode = Schema.RequiredMode.REQUIRED)
    private String firstName;

    @NotBlank(message = "Last name cannot be blank.")
    @Schema(description = "Last name", requiredMode = Schema.RequiredMode.REQUIRED)
    private String lastName;

    @NotBlank(message = "Job title cannot be blank.")
    @Schema(description = "Job title", requiredMode = Schema.RequiredMode.REQUIRED)
    private String jobTitle;

    @NotBlank(message = "Phone number cannot be blank.")
    @Schema(description = "Phone number", requiredMode = Schema.RequiredMode.REQUIRED)
    private String phoneNumber;
}

File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/OnboardingUserUpdateDto.kt
/*
 * Copyright (c) 2024 sovity GmbH
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contributors:
 *      sovity GmbH - initial implementation
 */

package de.sovity.authorityportal.api.model

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank

@Schema(description = "Information about the user provided while onboarding.")
data class OnboardingUserUpdateDto(
    @field:NotBlank(message = "First name cannot be blank.")
    @field:Schema(description = "First name", requiredMode = Schema.RequiredMode.REQUIRED)
    val firstName: String,

    @field:NotBlank(message = "Last name cannot be blank.")
    @field:Schema(description = "Last name", requiredMode = Schema.RequiredMode.REQUIRED)
    val lastName: String,

    @field:NotBlank(message = "Job title cannot be blank.")
    @field:Schema(description = "Job title", requiredMode = Schema.RequiredMode.REQUIRED)
    val jobTitle: String,

    @field:NotBlank(message = "Phone number cannot be blank.")
    @field:Schema(description = "Phone number", requiredMode = Schema.RequiredMode.REQUIRED)
    val phoneNumber: String,
)
